packages/macros/tests/babel/dependency-satisfies.test.ts

```diff
@@ -134,13 +134,39 @@ describe(`dependencySatisfies`, function () {
         expect(runDefault(code)).toBe(true);
       });
 
+      test(`it considers a project's used pre-releases as allowed`, () => {
+        project.addDependency('foo', '4.2.0-beta.1');
+        let code = transform(
+          `
+          import { dependencySatisfies } from '@embroider/macros';
+          export default function() {
+            return dependencySatisfies('foo', '>= 3.27.0-alpha.1 || >= 3.27.0-beta.1');
+          }
+        `
+        );
+        expect(runDefault(code)).toBe(true);
+      });
+
+      test(`in monorepos, the * dep is commonly used to mean "latest"`, () => {
+        project.addDependency('foo', '*');
```

Comment:
This isn't quite right, the second arg should be an actual version number

Reply:
The `*` you are referring to (in the commit comment) is in a version range field (e.g. inside `dependencies` and `devDependencies`) **not** the top level `version` field of a `package.json` (which is what this second arg is all about).

Reply:
oh :thinking: the nomenclature here, `project.addDependency` is telling me that, _in the consuming project_, I'd have:

```
"dependencies": {
  "foo": "*"
}
```

so, for extra clarification, if that's _not_ the case, I'll go ahead and make your suggested change, no questions asked.

What is a "project"? 

Fwiw, _this is the piece of code I'm trying to "get around": https://github.com/embroider-build/embroider/blob/main/packages/util/addon/ember-private-api.js#L11
Where, I feel like if the consuming project has either `*` or `<whatever current canary is>`, this block should be true. (in the released version of embroider, that's not the case).

Is there some other way to test this scenario?